src/preferences/pages/blacklist.ts

```diff
@@ -1,111 +1,73 @@
-                .get()
-                .connect(
-                    this._black_list_group,
-                    'row-activated',
-                    (_me: Gtk.ListBox, row: Gtk.ListBoxRow) => {
-                        if (row instanceof AppRow) {
-                            row.on_expanded_changed();
-                        }
-                    },
-                );
-        }
-
-        private show_err_msg(item: string) {
-            const title = `${item}: ${_(
-                "Can't add to list, because it has exists",
-            )}`;
-            show_err_msg(title);
         }
 
-        // --------------------------------------------------- [signal handlers]
-
-        private on_delete_row(row: Gtk.ListBoxRow, title: string) {
-            this.black_list.splice(this.black_list.indexOf(title), 1);
-            settings().black_list = this.black_list;
-            this._black_list_group.remove(row);
-        }
-
-        /** Add a row to black list */
-        private on_add_row(title = '') {
-            const handlers: AppRowHandler = {
-                on_delete: (row, title) => this.on_delete_row(row, title),
-                on_title_changed: (old_title, new_title) =>
-                    this.on_title_changed(old_title, new_title),
+        private add_window(_?: Gtk.Button, title?: string) {
+            const cb: AppRowCb = {
+                on_delete: row => this.delete_row(row),
+                on_title_changed: (_, old_title, new_title) =>
+                    this.change_title(old_title, new_title),
             };
 
-            const row = new AppRow(handlers);
-            row.title = title;
-
-            if (!title) {
-                row.description = TIPS_EMPTY();
-            }
+            const row = new BlacklistRow(cb);
+            row.set_subtitle(title || '');
+            this._blacklist_group.add(row);
+        }
 
-            this._black_list_group.append(row);
+        private delete_row(row: AppRowClass) {
+            this.blacklist.splice(this.blacklist.indexOf(row.title), 1);
+            settings().black_list = this.blacklist;
+            this._blacklist_group.remove(row);
         }
 
-        /** Called when title of item need to changed  */
-        on_title_changed(old_title: string, new_title: string): boolean {
-            if (this.black_list.includes(new_title)) {
-                this.show_err_msg(new_title);
+        private change_title(old_title: string, new_title: string): boolean {
+            if (this.blacklist.includes(new_title)) {
+                const win = this.root as unknown as Adw.PreferencesDialog;
+                win.add_toast(
+                    new Adw.Toast({
+                        title: gettext(
+                            `Can't add ${new_title} to the list, because it already there`,
+                        ),
+                    }),
+                );
                 return false;
             }
+            if (old_title === '') {
```

Comment:
```suggestion
            }
            
            if (old_title === '') {
```

(Just a matter of personal preference though)